hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java

```diff
@@ -389,7 +389,7 @@ private void getNextKey() throws IOException {
         Table.KeyValue<String, Value> entry = tableIterator.next();
         String entryKey = entry.getKey();
         if (entryKey.startsWith(prefixKey)) {
-          if (!KeyManagerImpl.isKeyDeleted(entryKey, table)) {
+          if (table.isExist(entryKey)) {
```

Comment:
getNextKey() call is on every hasNextCall(), so again here we are iterating on table and checking existence  of key, so is it acceptable from performance point of view ? One more point that as per isExist API doc:

`A lock on the key / bucket needs to be acquired before invoking this API.`